core/src/robot/AbstractConstraint.cc

```diff
@@ -0,0 +1,46 @@
+#include "jiminy/core/robot/Robot.h"
+#include "jiminy/core/Utilities.h"
+
+#include "jiminy/core/robot/AbstractConstraint.h"
+
+
+namespace jiminy
+{
+    AbstractConstraint::AbstractConstraint() :
+    model_(nullptr),
+    isAttached_(false),
+    jacobian_(matrixN_t::Zero(0,0)),
+    drift_(vectorN_t::Zero(0))
+    {
+        // Empty on purpose
+    }
+
+    matrixN_t const & AbstractConstraint::getJacobian(vectorN_t const & q)
+    {
+        return jacobian_;
+    }
+
+    vectorN_t const & AbstractConstraint::getDrift(vectorN_t const & q, vectorN_t const & v)
+    {
+        return drift_;
+    }
+
+
+    hresult_t AbstractConstraint::attach(Model const * model)
+    {
+        model_ = model;
+        isAttached_ = true;
+        return hresult_t::SUCCESS;
+    }
+
+    void AbstractConstraint::detach()
+    {
+        model_ = nullptr;
+        isAttached_ = false;
+    }
+
+    hresult_t AbstractConstraint::refreshProxies()
```

Comment:
.

Reply:
Because binding a pure virtual function requires extra work so as not to get an undefined symbol at library loading. I took the lazy option to make a concrete implementation instead.

Reply:
Sure, but this method should not have binding since it is only used internally. It is only public to avoid complex interface architecture.

Reply:
Only methodq useful for the user should be binded in Python.

Reply:
Even without binding the method, the simple fact that the class is binded poses a problem - because it means that the user can instanciate the class, but the method itself is undefined.

Reply:
Oh ok I get it - I though not having a definition would make the method pure virtual, but no you need to explicitly at ```= 0``` to the definition.  And you need to prvent python initialization with bp::no_init